File: 560. Subarray Sum Equals K/subarraySum.cpp
class Solution
{
public:
    int subarraySum(vector<int> &nums, int k)
    {

        unordered_map<int, int> m;
        int preSum = 0, cnt = 0;
        for (auto it : nums)
        {
            preSum += it;
            if (preSum == k)
            {
                cnt++;
            }
            if (m.find(preSum - k) != m.end())
            {
                cnt += m[preSum - k];
            }
            m[preSum]++;
        }
        return cnt;
    }
};

File: 560. Subarray Sum Equals K/subarraySum.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int preSum = 0, cnt = 0;

        for (int num : nums) {
            preSum += num;

            if (preSum == k) {
                cnt++;
            }

            if (map.containsKey(preSum - k)) {
                cnt += map.get(preSum - k);
            }

            map.put(preSum, map.getOrDefault(preSum, 0) + 1);
        }

        return cnt;
    }
}